bazel/repositories_extra.bzl

```diff
@@ -24,7 +24,7 @@ def envoy_dependencies_extra(
     python_register_toolchains(
         name = "python%s" % _python_minor_version(python_version),
         python_version = python_version,
-        ignore_root_user_error = ignore_root_user_error,
+        ignore_root_user_error = True,
```

Comment:
i think this is not necessary/desirable - at least not in the general case

Reply:
Its for this issue -
Error in fail: The current user is root, please run as non-root when using the hermetic Python interpreter. See https://github.com/bazelbuild/rules_python/pull/713.

Reply:
I agree about not changing this for the general case. However, I'm not sure how we can make the value of `ignore_root_user_error` depend on the target architecture.

I tried using a `select()` in the top level WORKSPACE file, to pass an `ignore_root_user_error` value...
```
envoy_dependencies_extra(ignore_root_user_error = select({
    "//bazel:linux_ppc": True,
    "//conditions:default": False,
}))
```
...but bazel does not allow the use of `select()` in that context.

Reply:
mostly this is for containers (altho it would also affect running as root in any other type of host) - is it not possible to run as non-root?

this setting has caused a lot annoyance (i think kinda justifiably) but we (envoy) decided it was better not to water it down as it does provide better security in most contexts - afaiaa there is not an easy workaround/override other than just not calling it in the WORKSPACE and doing the `rules_python` setup directly

if its not possible to run as non-root, and is required to do so, then maybe its necessary to set it here.

Reply:
re absence of `select` in workspace, i think you can do conditional stuff from the workspace using repository_rules but its a bit convoluted

Reply:
I am looking into this, but it looks complicated.

Reply:
yeah - non-trivial i think

i had a look at the upstream issue again in rules_python - it seems its not _for_ containers - its for issues around writing `.pyc` files - there is some discussion there about using env vars to override - but tbh im not sure how exactly

Reply:
I got the above workaround for this upstream issue https://github.com/bazelbuild/rules_python/issues/1169 , i believe its the same you are also referring. But I am unable to find any alternate solution for this clearly.

Reply:
I agree that its not necessary to run as root in general. Will remove this changes